src/org/zaproxy/zap/extension/ascan/ActiveScan.java

```diff
@@ -289,7 +296,27 @@ public void run() {
             });
         }
 	}
-	
+
+    private void removeHistoryReference(HistoryReference remRef) {
+        if (View.isInitialised()) {

```

Comment:
The local ones should still be used to `invokeLater` the calls to `messagesTableModel`. So, instead of:
```Java
private void addHistoryReference(HistoryReference hRef) {
    if (View.isInitialised()) {
        addHistoryReferenceInEdt(hRef);
    }
}

private void addHistoryReferenceInEdt(final HistoryReference hRef) {
    if (EventQueue.isDispatchThread()) {
        messagesTableModel.addHistoryReference(hRef);
    } else {
        EventQueue.invokeLater(new Runnable() {

            @Override
            public void run() {
                addHistoryReference(hRef);
            }
        });
    }
}
```
we just need:
```Java
private void addHistoryReferenceInEdt(final HistoryReference hRef) {
    EventQueue.invokeLater(new Runnable() {

        @Override
        public void run() {
            messagesTableModel.addHistoryReference(hRef);
        }
    });
}
```